agent/consul/state/memdb_wrapper.go

```diff
@@ -0,0 +1,92 @@
+package state
+
+import (
+	"github.com/hashicorp/go-memdb"
+)
+
+// memDBWrapper is a thin wrapper around memdb.DB which enables TrackChanges on
+// all write transactions. When the transaction is committed the changes are
+// sent to the eventPublisher which will create and emit change events.
+type memDBWrapper struct {
```

Comment:
I'm tempted to call this `changeTrackerDB` so that the name is a little more meaningful. Thoughts? It is not used in many places (6), so the name probably isn't too important.

Reply:
`changeTrackerDB` seems better to me. 👍